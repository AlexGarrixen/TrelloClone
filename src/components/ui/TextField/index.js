import React, { forwardRef } from 'react';
import PropTypes from 'prop-types';
import clsx from 'clsx';

const TextField = forwardRef(
  (
    {
      onChange,
      value,
      className,
      disabled,
      textArea = false,
      fullWidth = false,
      type = 'text',
      ...other
    },
    ref
  ) => {
    let Component = 'input';

    if (textArea) Component = 'textarea';

    return (
      <Component
        ref={ref}
        className={clsx('textfield', fullWidth && 'w-full', className)}
        onChange={onChange}
        value={value}
        type={type}
        {...other}
      />
    );
  }
);

TextField.propTypes = {
  onChange: PropTypes.func,
  value: PropTypes.string,
  disabled: PropTypes.bool,
  textArea: PropTypes.bool,
  type: PropTypes.string,
};

export default TextField;
