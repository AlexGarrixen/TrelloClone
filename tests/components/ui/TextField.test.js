import React from 'react';
import { mount } from 'enzyme';
import TextField from '../../../src/components/ui/TextField';

describe('<TextField />', () => {
  test('Render Success', () => {
    const wrapper = mount(<TextField />);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('.textfield').exists()).toBe(true);
  });

  test('Test with prop textArea', () => {
    const wrapper = mount(<TextField textArea />);
    const element = wrapper.find('.textfield');

    expect(element.type()).toBe('textarea');
  });

  test('Test with prop onChange', () => {
    const onChange = jest.fn();
    const wrapper = mount(<TextField onChange={onChange} />);
    const element = wrapper.find('input');

    element.simulate('change', { target: { value: 'test' } });

    expect(onChange).toHaveBeenCalled();
    expect(onChange).toHaveBeenCalledWith(expect.any(Object));
  });

  test('Test with prop value', () => {
    const wrapper = mount(<TextField value='test' onChange={() => {}} />);
    const element = wrapper.find('input');

    expect(element.prop('value')).toBe('test');
  });

  test('Test with prop fullWidth', () => {
    const wrapper = mount(<TextField fullWidth />);
    const element = wrapper.find('input');

    expect(element.hasClass('w-full')).toBe(true);
  });
});
