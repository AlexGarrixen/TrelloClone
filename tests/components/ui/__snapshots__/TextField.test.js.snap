// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextField /> Render Success 1`] = `
<input
  className="textfield"
  type="text"
/>
`;
